sys/kern/build.rs

```diff
@@ -153,175 +179,119 @@ fn generate_statics() -> Result<(), Box<dyn std::error::Error>> {
-        }
-    };
-    let fmt_task_irq = |v: Option<&(InterruptOwner, Vec<InterruptNum>)>| {
-        match v {
-            Some((owner, irqs)) => format!(
-                "(abi::InterruptOwner {{ task: {}, notification: 0b{:b} }}, &[{}]),",
-                owner.task, owner.notification,
-                irqs.iter()
-                    .map(|i| format!("abi::InterruptNum({})", i.0))
-                    .collect::<Vec<_>>()
-                    .join(", ")
-            ),
-            None => {
-                "(abi::InterruptOwner::invalid(), &[]),"
-                    .to_string()
-            }
-        }
-    };
-
     let target = build_util::target();
     if target.starts_with("thumbv6m") {
-        let task_irq_map =
-            phash_gen::OwnedSortedList::build(task_irq_map).unwrap();
-        let irq_task_map =
-            phash_gen::OwnedSortedList::build(irq_task_map).unwrap();
+        let task_irq_map = phash_gen::OwnedSortedList::build(task_irq_map)
+            .context("building task-to-IRQ map")?;
+        let irq_task_map = phash_gen::OwnedSortedList::build(irq_task_map)
+            .context("building IRQ-to-task map")?;
 
         // Generate text for the Interrupt and InterruptSet tables stored in the
-        // PerfectHashes
-        let irq_task_value = irq_task_map
-            .values
-            .iter()
-            .map(|o| fmt_irq_task(Some(o)))
-            .collect::<Vec<String>>()
-            .join("\n        ");
-        let task_irq_value = task_irq_map
-            .values
-            .iter()
-            .map(|o| fmt_task_irq(Some(o)))
-            .collect::<Vec<String>>()
-            .join("\n        ");
-
-        write!(file, "
-use phash::SortedList;
-pub const HUBRIS_IRQ_TASK_LOOKUP: SortedList::<abi::InterruptNum, abi::InterruptOwner> = SortedList {{
-    values: &[
-        {}
-    ],
-}};
-pub const HUBRIS_TASK_IRQ_LOOKUP: SortedList::<abi::InterruptOwner, &'static [abi::InterruptNum]> = SortedList {{
-    values: &[
-        {}
-    ],
-}};",
-        irq_task_value, task_irq_value)?;
+        // sorted lists
+        let irq_task_literal = fmt_sorted_list(&irq_task_map, fmt_irq_task);
+        let task_irq_literal = fmt_sorted_list(&task_irq_map, fmt_task_irq);
+
+        write!(
+            file,
+            "{}",
+            quote::quote! {
+                pub const HUBRIS_IRQ_TASK_LOOKUP:
+                    phash::SortedList<abi::InterruptNum, abi::InterruptOwner>
+                    = #irq_task_literal;
+                pub const HUBRIS_TASK_IRQ_LOOKUP:
+                    phash::SortedList<
+                        abi::InterruptOwner,
+                        &'static [abi::InterruptNum],
+                    > = #task_irq_literal;
+            }
+        )?;
     } else if target.starts_with("thumbv7m")
         || target.starts_with("thumbv7em")
         || target.starts_with("thumbv8m")
     {
-        let nested_import = if let Ok(task_irq_map) =
+        writeln!(
```

Comment:
It looks like `phash` members are now always fully qualified – can this import be removed entirely?

Reply:
It appears so!